core/src/main/java/com/linecorp/armeria/internal/HttpHeaderSubscriber.java

```diff
@@ -63,7 +68,6 @@ public void onNext(HttpObject o) {
             final HttpStatus status = headers.status();
             if (status != null && status.codeClass() != HttpStatusClass.INFORMATIONAL && this.headers == null) {
                 this.headers = headers;
-                subscription.cancel();
```

Comment:
Just curious. Is it okay to remove `subscription.cancel()` here? 
I'm not sure, but is there any chance that `HttpData` is passed here due to the missing of cancellation?
If so, don't we need to release `HttpData` here if it is `ByteBufHttpData`?

Reply:
I designed to use this class like `HttpMessageAggregator` which does `subscribe(false /* withPooledObject */)`, so do not have to worry about releasing the reference. However. this is a `public` class(even though it is in the `internal` package), I think I need to explain about it in the javadoc.

I was not sure about the `subscription.cancel()`, but it might be better to put it back.

Reply:
Got it. Thank you for explanation. :-)